packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -0,0 +1,327 @@
+      return BillingResultWrapper(responseCode: BillingResponse.ok);
+    }
+
+    if (googlePurchase.verificationData == null) {
+      throw ArgumentError(
+          'completePurchase unsuccessful. The `purchase.verificationData` is not valid');
+    }
+
+    return await billingClient
+        .acknowledgePurchase(purchase.verificationData.serverVerificationData);
+  }
+
+  /// Mark that the user has consumed a product.
+  ///
+  /// You are responsible for consuming all consumable purchases once they are
+  /// delivered. The user won't be able to buy the same product again until the
+  /// purchase of the product is consumed.
+  Future<BillingResultWrapper> consumePurchase(PurchaseDetails purchase) {
+    if (purchase.verificationData == null) {
+      throw ArgumentError(
+          'consumePurchase unsuccessful. The `purchase.verificationData` is not valid');
+    }
+    return billingClient
+        .consumeAsync(purchase.verificationData.serverVerificationData);
+  }
+
+  @override
+  Future<void> restorePurchases({
+    String? applicationUserName,
+  }) async {
+    List<PurchasesResultWrapper> responses;
+
+    responses = await Future.wait([
+      billingClient.queryPurchases(SkuType.inapp),
+      billingClient.queryPurchases(SkuType.subs)
+    ]);
+
+    Set errorCodeSet = responses
+        .where((PurchasesResultWrapper response) =>
+            response.responseCode != BillingResponse.ok)
+        .map((PurchasesResultWrapper response) =>
+            response.responseCode.toString())
+        .toSet();
+
+    String errorMessage =
+        errorCodeSet.isNotEmpty ? errorCodeSet.join(', ') : '';
+
+    List<PurchaseDetails> pastPurchases =
+        responses.expand((PurchasesResultWrapper response) {
+      return response.purchasesList;
+    }).map((PurchaseWrapper purchaseWrapper) {
+      return GooglePlayPurchaseDetails.fromPurchase(purchaseWrapper);
+    }).toList();
+
+    if (errorMessage.isNotEmpty) {
+      throw InAppPurchaseException(
+        source: kIAPSource,
+        code: kRestoredPurchaseErrorCode,
+        message: errorMessage,
+      );
+    }
+
+    _purchaseUpdatedController.add(pastPurchases);
+  }
+
+  /// Resets the connection instance.
+  ///
+  /// The next call to [instance] will create a new instance. Should only be
+  /// used in tests.
+  @visibleForTesting
+  static void reset() => _instance = null;
+
+  static InAppPurchaseAndroidPlatform _getOrCreateInstance() {
+    if (_instance != null) {
+      return _instance!;
+    }
+
+    _instance = InAppPurchaseAndroidPlatform._();
+    return _instance!;
+  }
+
+  Future<void> _connect() =>
+      billingClient.startConnection(onBillingServiceDisconnected: () {});
+
+  /// Query the product detail list.
+  ///
+  /// This method only returns [ProductDetailsResponse].
+  /// To get detailed Google Play sku list, use [BillingClient.querySkuDetails]
+  /// to get the [SkuDetailsResponseWrapper].
+  Future<ProductDetailsResponse> queryProductDetails(
```

Comment:
Where is this method used? This is an addition API right? Which should be in the addition class?

Reply:
This method is declared in the `InAppPurchasePlatform` plugin interface, so it should have been decorated with the `@override` attribute. I corrected this mistake.